### PR TITLE
Fix and move helm base dir

### PR DESCRIPTION
**Pull request by zhu733756**
**What type of PR is this?**
Fix and move helm base dir.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #https://github.com/kubeedge/kubeedge/issues/3660

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
helm base dir changed
```

### CODE REVIEW

File: keadm/cmd/keadm/app/cmd/util/helminstaller.go [hunk 1]
```diff
@@ -381,15 +381,9 @@ func (cu *KubeCloudHelmInstTool) handleProfile(profileValue string) error {
...
 		}
-		profileValueSuffix := strings.TrimPrefix(profileValue, "v")
-		// confirm it startswith "v"
-		if profileValue != profileValueSuffix {
-			cu.Sets = append(cu.Sets, fmt.Sprintf("%s=v%s", "cloudCore.image.tag", profileValueSuffix))
-			cu.Sets = append(cu.Sets, fmt.Sprintf("%s=v%s", "iptablesManager.image.tag", profileValueSuffix))
-		} else {
-			cu.Sets = append(cu.Sets, fmt.Sprintf("%s=%s", "cloudCore.image.tag", profileValue))
-			cu.Sets = append(cu.Sets, fmt.Sprintf("%s=%s", "iptablesManager.image.tag", profileValue))
-		}
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/util/helminstaller.go:392]: v is no need now?

File: keadm/cmd/keadm/app/cmd/util/helminstaller.go [hunk 2]
```diff
@@ -18,8 +18,8 @@ import (
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 	"sigs.k8s.io/yaml"
 
-	keCharts "github.com/kubeedge/kubeedge/build/helm/charts"
 	"github.com/kubeedge/kubeedge/common/constants"
+	keCharts "github.com/kubeedge/kubeedge/helm/charts"
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/util/helminstaller.go:22]: For import, it's better to use the lowcase?

File: keadm/cmd/keadm/app/cmd/util/helminstaller.go (hunk 1 above)
Comment (zhu733756) [reply keadm/cmd/keadm/app/cmd/util/helminstaller.go:392]: Yeah, this is a custom flag,  "cloudCore.image.tag" or "iptablesmgr.image.tag" is controlled by user if they have a custom image. If not given, will use the default kubeedge version.

File: keadm/cmd/keadm/app/cmd/util/helminstaller.go (hunk 2 above)
Comment (zhu733756) [reply keadm/cmd/keadm/app/cmd/util/helminstaller.go]: Done.

File: keadm/cmd/keadm/app/cmd/util/helminstaller.go (hunk 1 above)
Comment (fisherxu) [reply keadm/cmd/keadm/app/cmd/util/helminstaller.go:392]: Okey, users can set the custom value, so if we always add the `v`, it not reasonable, right?

Comment (zhu733756) [reply keadm/cmd/keadm/app/cmd/util/helminstaller.go:392]: > Okey, users can set the custom value, so if we always add the `v`, it not reasonable, right?

I think using v is professional, but once the user sets a custom image tag, should override our default version.

Comment (fisherxu) [reply keadm/cmd/keadm/app/cmd/util/helminstaller.go:392]: In k8s/kubeadm, it has a version match, if match successfully, will add  the `v`. If match failed, which should means the version is customed, will not add `v`.

I think users may set the version but ignore the `v` prefix, so deploy will failed, how about we also  have a check like k8s? @zhu733756

Comment (fisherxu) [reply keadm/cmd/keadm/app/cmd/util/helminstaller.go:392]: Ref: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/util/version.go#L152

Comment (zhu733756) [reply keadm/cmd/keadm/app/cmd/util/helminstaller.go:392]: Seems good.

Comment (zhu733756) [reply keadm/cmd/keadm/app/cmd/util/helminstaller.go:392]: But I am worried that if the user gives a tag that matched the regex but is not given a "v" prefix string,  their custom image tag would not match through this method when we return a "v" prefix.

File: keadm/cmd/keadm/app/cmd/util/helminstaller.go [hunk 3]
```diff
@@ -205,17 +207,21 @@ func (cu *KubeCloudHelmInstTool) beforeRenderer(baseHelmRoot string) error {
...
-		if cu.CloudcoreImage != "" {
-			cu.Sets = append(cu.Sets, fmt.Sprintf("%s=%s", "cloudCore.image.repository", cu.CloudcoreImage))
+
+		if cu.CloudcoreRepo != "" {
+			cu.Sets = append(cu.Sets, fmt.Sprintf("%s=%s", "cloudCore.image.repository", cu.CloudcoreRepo))
 		}
+
 		if cu.CloudcoreTag != "" {
-			cu.Sets = append(cu.Sets, fmt.Sprintf("%s=%s", "cloudCore.image.tag", cu.CloudcoreTag))
+			cu.Sets = append(cu.Sets, fmt.Sprintf("%s=%s", "cloudCore.image.tag", normalizedBuildVersion(cu.CloudcoreTag)))
```
Comment (zhu733756) [inline keadm/cmd/keadm/app/cmd/util/helminstaller.go:216]: We suggest the custom image build with a v prefix.

**Review by fisherxu — COMMENTED** [review]
/approve
/assign @zc2638

**Comment by zc2638** [comment]
I recommend getting a unified version everywhere. #3643
And it is not recommended to specify a specific component image repo like `CloudcoreRepo`, otherwise not only the maintenance cost is high, but the user cost is also high

**Comment by zc2638** [comment]
There may be other resources that need to be embedded in the future. It is recommended to unify a reasonable directory and place the helm manifest resources in the secondary directory.
@fisherxu

**Comment by zhu733756** [comment]
> I recommend getting a unified version everywhere. #3643 

This pr is a little different from getting the default kubeedge version, it is actually a image tag, user can define it as what they wanted. 

> And it is not recommended to specify a specific component image repo like `CloudcoreRepo`, otherwise not only the maintenance cost is high, but the user cost is also high

It comes from the common helm values definition,  repo seems to repository, always described as "kubeedge/cloudcore", and tag means the tag of this image.  See https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml#L415

**Comment by zhu733756** [comment]
> There may be other resources that need to be embedded in the future. It is recommended to unify a reasonable directory and place the helm manifest resources in the secondary directory. @fisherxu

> I recommend getting a unified version everywhere. #3643 

The regex should move to a common position I think.

> And it is not recommended to specify a specific component image repo like `CloudcoreRepo`, otherwise not only the maintenance cost is high, but the user cost is also high

It comes from the common helm values definition. See https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/values.yaml#L415. It is a fast path for set flag like "--set cloudCore.image.repo=xxx".  Do we unexpose it?

**Comment by zhu733756** [comment]
```
1._output/local/bin/keadm beta init --advertise-address=192.168.88.6 --set cloudCore.service.enable=true --set iptablesManager.mode=external --set cloudCore.hostNetWork=false --kubeedge-version=v1.10.0 --force
2. _output/local/bin/keadm beta init --advertise-address=192.168.88.6 --set cloudCore.service.enable=true --set iptablesManager.mode=external --set cloudCore.hostNetWork=false --profile version=1.10.0 --force
3.  _output/local/bin/keadm beta init --advertise-address=192.168.88.6 --set cloudCore.service.enable=true --set iptablesManager.mode=external --set cloudCore.hostNetWork=false --profile version=v1.10.0 --force
```

Neither one of these above three commands works well.

File: keadm/cmd/keadm/app/cmd/beta/manifest.go [hunk 4]
```diff
@@ -95,62 +96,55 @@ func NewBetaManifestGenerate() *cobra.Command {
...
-	if BetaInitOptions.Namespace == "" {
-		BetaInitOptions.Namespace = constants.SystemNamespace
+		kubeedgeVersion = types.DefaultKubeEdgeVersion
+		fmt.Println("Failed to get the latest KubeEdge release version, will use default version: ", kubeedgeVersion)
 	}
 
 	common := util.Common{
-		ToolVersion: semver.MustParse(kubeVer),
-		KubeConfig:  BetaInitOptions.KubeConfig,
+		ToolVersion: semver.MustParse(kubeedgeVersion),
```
Comment (zhu733756) [inline keadm/cmd/keadm/app/cmd/beta/manifest.go:129]: kubeedgeVersion would use a unified method to generate in the future.

**Comment by zhu733756** [comment]
Pls move to #3692.  Emmm, I found that if the user doesn't have a remote tag v1.10.0+, the e2e test would be failed.

**Pull request closed without merging**